/*

   calin/proto/iact_data/zfits_data_source.proto -- Stephen Fegan -- 2016-01-21

   Protobufs for configuration of ZFITSDataSource

   Copyright 2016, Stephen Fegan <sfegan@llr.in2p3.fr>
   LLR, Ecole polytechnique, CNRS/IN2P3, Universite Paris-Saclay

   This file is part of "calin"

   "calin" is free software: you can redistribute it and/or modify it
   under the terms of the GNU General Public License version 2 or
   later, as published by the Free Software Foundation.

   "calin" is distributed in the hope that it will be useful, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   General Public License for more details.

*/

syntax = "proto3";

import "calin.proto";

package calin.ix.iact_data.zfits_data_source;

message ZFITSDataSourceConfig {
  string extension                                         = 1 [
    (CFO).desc = "Standard extension for ZFits filenames, used for building "
      "chained data file names." ];
  bool log_on_file_open                                    = 2 [
    (CFO).desc = "Send a message to the log when opening a file." ];
  bool exact_filename_only                                 = 3 [
    (CFO).desc = "Only try to open file with given (exact) filename." ];
  uint32 max_file_fragments                                = 4 [
    (CFO).desc = "Maximum number of file fragments to open. Zero means open "
      "all fragments found." ];
  bool dont_read_run_header                                = 11 [
    (CFO).desc = "Don't read the run header." ];
  bool ignore_run_header_errors                            = 12 [
    (CFO).desc = "Don't log missing run header." ];
};
